include/seqan3/io/detail/misc_input.hpp

```diff
@@ -104,14 +104,14 @@ inline auto make_secondary_istream(std::basic_istream<char_t> & primary_stream,
         extension = filename.extension().string().substr(1);
 
     // tests whether the given extension matches with one of the given compression tags.
-    auto contains_extension = [] (auto compression_tag, auto const & extension)
+    [[maybe_unused]] auto contains_extension = [] (auto compression_tag, auto const & extension)
```

Comment:
why don't you move this inside of the first if scope to avoid the maybe_unused macro?

Reply:
That does not work. Since it might be that zlib is not available but zstd is.